packages/auto_mappr/test/integration/fixture/type_converter.dart

```diff
@@ -0,0 +1,124 @@
+import 'package:auto_mappr_annotation/auto_mappr_annotation.dart';
+import 'package:equatable/equatable.dart';
+
+part 'type_converter.g.dart';
+
+class DateDomain {
+  final DateTime? dateTimeA;
+  final DateTime? dateTimeB;
+  final DateTime? dateTimeC;
+
+  DateDomain({
+    required this.dateTimeA,
+    required this.dateTimeB,
+    required this.dateTimeC,
+  });
+}
+
+class DateDto {
+  final String dateTimeA;
+  final String dateTimeB;
+  final String dateTimeC;
+
+  DateDto({
+    required this.dateTimeA,
+    required this.dateTimeB,
+    required this.dateTimeC,
+  });
+}
+
+class DateDto2 {
+  final String dateTimeA;
+  final String dateTimeB;
+
+  DateDto2({
+    required this.dateTimeA,
+    required this.dateTimeB,
+  });
+}
+
+class UserId extends Equatable {
+  final String value;
+
+  @override
+  List<Object?> get props => [value];
+
+  const UserId(this.value);
+}
+
+class AccountId extends Equatable {
+  final String value;
+
+  @override
+  List<Object?> get props => [value];
+
+  const AccountId(this.value);
+}
+
+class User {
+  final UserId? id;
+  final AccountId? accountId;
+
+  User(this.id, this.accountId);
+}
+
+class UserDto {
+  final String? id;
+  final String? accountId;
+
+  UserDto(this.id, this.accountId);
+}
+
+
+@AutoMappr(
+  [
+    MapType<DateDto, DateDomain>(
+      fields: [
+        Field(
+          'dateTimeA',
+          type: TypeConverter<String, DateTime>(Mappr.tryParseFirstOfDecemberInYear),
+        ),
+      ],
+      types: [
+        TypeConverter<String, DateTime>(Mappr.parseSecondOfDecemberInYear),
+      ],
+    ),
+    MapType<DateDto2, DateDomain>(),
+    MapType<UserDto, User>(
+      fields: [
+        Field('id', type: TypeConverter<String, UserId>(UserId.new)),
+        Field('accountId', type: TypeConverter<String, AccountId>(AccountId.new)),
+      ],
+    ),
+  ],
+  types: [
+    TypeConverter<String, DateTime>(Mappr.parseISO8601),
+  ],
+)
```

Comment:
Is there any benefit of doing `TypeConverter<String, AccountId>(SomeFunction.xxx)` instead of just passing `SomeFunction.xxx`?

Reply:
Actually you need the explicit definition because inference doesn't seem to work without it. The wrapper Object is there to extend it with further configuration options.

Reply:
Not sure right now, but isn't it similar to `custom` where it also works for functions?

Reply:
You're right, it would probably work without the wrapper object. But then we'd lose the option to configure the TypeConverter further with e.g. field name constraints.

For brevity we could support both passing just the function or the whole Object I guess. What do you think?